runtime/compiler/control/CompilationThread.cpp

```diff
@@ -9122,15 +9256,72 @@ TR::CompilationInfo::compilationEnd(J9VMThread * vmThread, TR::IlGeneratorMethod
 
                   if (safeToStore)
                      {
+                     const U_8 *metadataToStore = dataStart;
+                     const U_8 *codedataToStore = codeStart;
+                     int metadataToStoreSize = dataSize;
+                     int codedataToStoreSize = codeSize;
+#ifdef COMPRESS_AOT_DATA
+                     if (!comp->getOption(TR_DisableAOTBytesCompression))
+                        {
+                        void * uncompressedData = comp->trMemory()->allocateHeapMemory(codeSize+dataSize);
+                        void * compressedData = comp->trMemory()->allocateHeapMemory(codeSize+dataSize);
+
+                        // Storing the uncompressed metadata in both buffer. 
+                        memcpy(uncompressedData, dataStart, dataSize);
+                        memcpy((U_8*)(uncompressedData)+dataSize, codeStart, codeSize);
+                        int uncompressedDataSize = sizeof(J9JITDataCacheHeader)+sizeof(TR_AOTMethodHeader);
```

Comment:
This should be renamed to `uncompressedHeaderSize` or something; it's confusing below because you have `dataSize` and `uncompressedDataSize` and `compressedDataSize`.

Reply:
Renamed the variables to reflect their use within the function in https://github.com/eclipse/openj9/pull/5926/commits/ed564d03add53c4f02ffb6e81bc3961a7aaddc1e